backend/test_gemini.py

```diff
@@ -0,0 +1,519 @@
+import pytest
+import prompts.rest_prompt
+from gemini import app, setup_llm, analyze_api_code, generate_basic_test
+
```

Comment:
```suggestion
from unittest.mock import patch, MagicMock
```

Using MagicMock would greatly simply the way you Mock each function and object. Instead of defining classes, and function for the tests, mocking assume the defined mocked behavior needed without the need to create the objects